nexus/tests/integration_tests/disks.rs

```diff
@@ -2389,6 +2389,83 @@ async fn test_single_region_allocate_for_replace_not_enough_zpools(
     assert_eq!(datasets_and_regions.len(), REGION_REDUNDANCY_THRESHOLD);
 }
 
+// Confirm that a region set can start at N, a region can be deleted, and the
+// allocation CTE can bring the redundancy back to N.
```

Comment:
Love to see it, this mitigates any fears I had about flexibility from the Nexus side!